build.gradle.kts

```diff
@@ -1,82 +1,95 @@
-
+  implementation("org.testcontainers:localstack")
+  implementation("org.testcontainers:postgresql")
+  implementation("org.testcontainers:testcontainers")
+  developmentOnly("org.springframework.boot:spring-boot-devtools")
   testImplementation("org.springframework.security:spring-security-test")
-  testImplementation("io.jsonwebtoken:jjwt-impl:0.12.3")
-  testImplementation("io.jsonwebtoken:jjwt-jackson:0.12.3")
-  testImplementation("org.mockito:mockito-inline:5.2.0")
-  testImplementation("io.mockk:mockk:1.13.11")
-  testImplementation("org.mockito.kotlin:mockito-kotlin:5.3.1")
-  testImplementation("org.testcontainers:localstack")
+  testImplementation("io.jsonwebtoken:jjwt-impl:0.12.5")
+  testImplementation("io.jsonwebtoken:jjwt-jackson:0.12.5")
+  testImplementation("org.jetbrains.kotlin:kotlin-test-junit5")
+  testRuntimeOnly("org.junit.platform:junit-platform-launcher")
   testImplementation("com.h2database:h2")
 }
 
-kotlin {
-  jvmToolchain(21)
-}
+testing {
+  suites {
+    val test by getting(JvmTestSuite::class) {
+      useJUnitJupiter()
+    }
 
-repositories {
-  mavenCentral()
+    register<JvmTestSuite>("integrationTest") {
+      dependencies {
+        testType.set(TestSuiteType.INTEGRATION_TEST)
+        kotlin.target.compilations { named("integrationTest") { associateWith(getByName("main")) } }
+        implementation("org.springframework.boot:spring-boot-starter-test")
+        implementation("org.springframework.security:spring-security-test")
+        implementation("io.jsonwebtoken:jjwt-impl:0.12.5")
+        implementation("io.jsonwebtoken:jjwt-jackson:0.12.5")
+        implementation("com.microsoft.azure:applicationinsights-logging-logback:2.6.4")
+        implementation("com.microsoft.azure:applicationinsights-logging-logback")
+        runtimeOnly("org.flywaydb:flyway-database-postgresql")
+        implementation("com.zaxxer:HikariCP:5.1.0")
+        implementation("com.h2database:h2")
+      }
+
+      targets {
+        all {
+          testTask.configure {
+            shouldRunAfter(test)
+          }
+        }
+      }
+    }
+  }
 }
 
 tasks {
   withType<KotlinCompile> {
-    compilerOptions.jvmTarget = JVM_21
-    compilerOptions.freeCompilerArgs = listOf("-Xjvm-default=all")
-  }
-}
+    named("check") {
+      dependsOn(testing.suites.named("integrationTest"))
+    }
 
-tasks.test {
-  finalizedBy(tasks.jacocoTestReport)
-}
+    named("compileIntegrationTestKotlin") {
+      dependsOn(named("copyAgent"))
+    }
+
+    named<JacocoReport>("jacocoTestReport") {
+      dependsOn(named("check"))
+      reports {
+        html.required.set(true)
+        xml.required.set(true)
+      }
+    }
 
-tasks.jacocoTestReport {
-  dependsOn(tasks.test)
-  reports {
-    html.required.set(true)
-    xml.required.set(true)
+    finalizedBy(named("jacocoTestReport"))
+    named("assemble") {
+      doFirst {
+        delete(
+          fileTree(project.layout.buildDirectory.get())
```

Comment:
I don't think this is needed? - we don't do this in other projects

Reply:
(something similar was previously incorrectly deleting the main app jar preventing deployments)

Reply:
(something similar was previously incorrectly deleting the main app jar preventing deployments)